src/services/expenseServices.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const { getUserById } = require('./userServices');
+
+let expenses = [];
+
+const init = () => {
+  expenses = [];
+};
+
+const validateData = (req) => {
+  const requiredFields = [
+    'userId',
+    'spentAt',
+    'title',
+    'amount',
+    'category',
+  ];
+
+  const requestedField = Object.keys(req);
+
+  const hasRequiredFields = requiredFields
+    .every(field => requestedField.includes(field));
+
+  const user = getUserById(req.userId);
+
+  if (!user || !hasRequiredFields || !requiredFields.length) {
```

Comment:
The condition '!requiredFields.length' will never be true, because 'requiredFields' is a constant array with elements.